### ISSUE TITLE
how  could i  get old version of libtorch , thanks

### ISSUE THREAD

**Issue by CaichaoGitHub** [pytorch/pytorch#40961]
## ❓ Questions and Help

hello, please help
my server is running a old version centos ,right now it can only support C++11, 
i wanted to install a old version of libtorch , but i can't find it anywhere ,

![image](https://user-images.githubusercontent.com/12059897/86428772-296f8900-bd20-11ea-921f-2646f526163e.png)

**Comment by peterjc123** [comment]
https://pytorch.org/get-started/previous-versions/

**Comment by peterjc123** [comment]
And the format of the download link for libtorch files is almost consistent. You can just replace 1.5.1 with any version you want.

**Comment by peterjc123** [comment]
For example, https://download.pytorch.org/libtorch/cu102/libtorch-shared-with-deps-1.4.0.zip

**Comment by CaichaoGitHub** [comment]
> For example,
> https://download.pytorch.org/libtorch/cu101/libtorch-shared-with-deps-1.4.0.zip

thanks very much , i have get it !

**Comment by jnulzl** [comment]
Hi @CaichaoGitHub and all others

# Linux libtorch 1.4.0 ~ 1.10.1 gpu version url

- cxx11 ABI


```shell
libtorch-cxx11-abi-shared-with-deps-1.4.0+cu92.zip : https://download.pytorch.org/libtorch/cu92/libtorch-cxx11-abi-shared-with-deps-1.4.0%2Bcu92.zip
libtorch-cxx11-abi-shared-with-deps-1.4.0+cu100.zip : https://download.pytorch.org/libtorch/cu100/libtorch-cxx11-abi-shared-with-deps-1.4.0%2Bcu100.zip
libtorch-cxx11-abi-shared-with-deps-1.4.0.zip : https://download.pytorch.org/libtorch/cu101/libtorch-cxx11-abi-shared-with-deps-1.4.0.zip
libtorch-cxx11-abi-shared-with-deps-1.5.0+cu92.zip : https://download.pytorch.org/libtorch/cu92/libtorch-cxx11-abi-shared-with-deps-1.5.0%2Bcu92.zip
libtorch-cxx11-abi-shared-with-deps-1.5.0+cu101.zip : https://download.pytorch.org/libtorch/cu101/libtorch-cxx11-abi-shared-with-deps-1.5.0%2Bcu101.zip
libtorch-cxx11-abi-shared-with-deps-1.5.0.zip : https://download.pytorch.org/libtorch/cu101/libtorch-cxx11-abi-shared-with-deps-1.5.0.zip
libtorch-cxx11-abi-shared-with-deps-1.5.0+cu102.zip : https://download.pytorch.org/libtorch/cu102/libtorch-cxx11-abi-shared-with-deps-1.5.0%2Bcu102.zip
libtorch-cxx11-abi-shared-with-deps-1.5.0.zip : https://download.pytorch.org/libtorch/cu102/libtorch-cxx11-abi-shared-with-deps-1.5.0.zip
libtorch-cxx11-abi-shared-with-deps-1.5.1+cu92.zip : https://download.pytorch.org/libtorch/cu92/libtorch-cxx11-abi-shared-with-deps-1.5.1%2Bcu92.zip
libtorch-cxx11-abi-shared-with-deps-1.5.1+cu101.zip : https://download.pytorch.org/libtorch/cu101/libtorch-cxx11-abi-shared-with-deps-1.5.1%2Bcu101.zip
libtorch-cxx11-abi-shared-with-deps-1.5.1.zip : https://download.pytorch.org/libtorch/cu102/libtorch-cxx11-abi-shared-with-deps-1.5.1.zip
libtorch-cxx11-abi-shared-with-deps-1.6.0+cu92.zip : https://download.pytorch.org/libtorch/cu92/libtorch-cxx11-abi-shared-with-deps-1.6.0%2Bcu92.zip
libtorch-cxx11-abi-shared-with-deps-1.6.0+cu101.zip : https://download.pytorch.org/libtorch/cu101/libtorch-cxx11-abi-shared-with-deps-1.6.0%2Bcu101.zip
libtorch-cxx11-abi-shared-with-deps-1.6.0.zip : https://download.pytorch.org/libtorch/cu102/libtorch-cxx11-abi-shared-with-deps-1.6.0.zip
libtorch-cxx11-abi-shared-with-deps-1.7.0+cu92.zip : https://download.pytorch.org/libtorch/cu92/libtorch-cxx11-abi-shared-with-deps-1.7.0%2Bcu92.zip
libtorch-cxx11-abi-shared-with-deps-1.7.0+cu101.zip : https://download.pytorch.org/libtorch/cu101/libtorch-cxx11-abi-shared-with-deps-1.7.0%2Bcu101.zip
libtorch-cxx11-abi-shared-with-deps-1.7.0.zip : https://download.pytorch.org/libtorch/cu102/libtorch-cxx11-abi-shared-with-deps-1.7.0.zip
libtorch-cxx11-abi-shared-with-deps-1.7.0+cu110.zip : https://download.pytorch.org/libtorch/cu110/libtorch-cxx11-abi-shared-with-deps-1.7.0%2Bcu110.zip
libtorch-cxx11-abi-shared-with-deps-1.7.1+cu92.zip : https://download.pytorch.org/libtorch/cu92/libtorch-cxx11-abi-shared-with-deps-1.7.1%2Bcu92.zip
libtorch-cxx11-abi-shared-with-deps-1.7.1+cu101.zip : https://download.pytorch.org/libtorch/cu101/libtorch-cxx11-abi-shared-with-deps-1.7.1%2Bcu101.zip
libtorch-cxx11-abi-shared-with-deps-1.7.1.zip : https://download.pytorch.org/libtorch/cu102/libtorch-cxx11-abi-shared-with-deps-1.7.1.zip
libtorch-cxx11-abi-shared-with-deps-1.7.1+cu110.zip : https://download.pytorch.org/libtorch/cu110/libtorch-cxx11-abi-shared-with-deps-1.7.1%2Bcu110.zip
libtorch-cxx11-abi-shared-with-deps-1.8.0+cu101.zip : https://download.pytorch.org/libtorch/cu101/libtorch-cxx11-abi-shared-with-deps-1.8.0%2Bcu101.zip
libtorch-cxx11-abi-shared-with-deps-1.8.0.zip : https://download.pytorch.org/libtorch/cu102/libtorch-cxx11-abi-shared-with-deps-1.8.0.zip
libtorch-cxx11-abi-shared-with-deps-1.8.0+cu111.zip : https://download.pytorch.org/libtorch/cu111/libtorch-cxx11-abi-shared-with-deps-1.8.0%2Bcu111.zip
libtorch-cxx11-abi-shared-with-deps-1.8.0+cu112.zip : https://download.pytorch.org/libtorch/cu112/libtorch-cxx11-abi-shared-with-deps-1.8.0%2Bcu112.zip
libtorch-cxx11-abi-shared-with-deps-1.8.1+cu101.zip : https://download.pytorch.org/libtorch/cu101/libtorch-cxx11-abi-shared-with-deps-1.8.1%2Bcu101.zip
libtorch-cxx11-abi-shared-with-deps-1.8.1+cu102.zip : https://download.pytorch.org/libtorch/cu102/libtorch-cxx11-abi-shared-with-deps-1.8.1%2Bcu102.zip
libtorch-cxx11-abi-shared-with-deps-1.8.1+cu111.zip : https://download.pytorch.org/libtorch/cu111/libtorch-cxx11-abi-shared-with-deps-1.8.1%2Bcu111.zip
libtorch-cxx11-abi-shared-with-deps-1.9.0+cu102.zip : https://download.pytorch.org/libtorch/cu102/libtorch-cxx11-abi-shared-with-deps-1.9.0%2Bcu102.zip
libtorch-cxx11-abi-shared-with-deps-1.9.0+cu111.zip : https://download.pytorch.org/libtorch/cu111/libtorch-cxx11-abi-shared-with-deps-1.9.0%2Bcu111.zip
libtorch-cxx11-abi-shared-with-deps-1.10.0+cu102.zip : https://download.pytorch.org/libtorch/cu102/libtorch-cxx11-abi-shared-with-deps-1.10.0%2Bcu102.zip
libtorch-cxx11-abi-shared-with-deps-1.10.0+cu111.zip : https://download.pytorch.org/libtorch/cu111/libtorch-cxx11-abi-shared-with-deps-1.10.0%2Bcu111.zip
libtorch-cxx11-abi-shared-with-deps-1.10.0+cu113.zip : https://download.pytorch.org/libtorch/cu113/libtorch-cxx11-abi-shared-with-deps-1.10.0%2Bcu113.zip
libtorch-cxx11-abi-shared-with-deps-1.10.1+cu102.zip : https://download.pytorch.org/libtorch/cu102/libtorch-cxx11-abi-shared-with-deps-1.10.1%2Bcu102.zip
libtorch-cxx11-abi-shared-with-deps-1.10.1+cu111.zip : https://download.pytorch.org/libtorch/cu111/libtorch-cxx11-abi-shared-with-deps-1.10.1%2Bcu111.zip
libtorch-cxx11-abi-shared-with-deps-1.10.1+cu113.zip : https://download.pytorch.org/libtorch/cu113/libtorch-cxx11-abi-shared-with-deps-1.10.1%2Bcu113.zip
```

- Pre-cxx11 ABI

```shell
libtorch-shared-with-deps-1.4.0+cu92.zip : https://download.pytorch.org/libtorch/cu92/libtorch-shared-with-deps-1.4.0%2Bcu92.zip
libtorch-shared-with-deps-1.4.0+cu100.zip : https://download.pytorch.org/libtorch/cu100/libtorch-shared-with-deps-1.4.0%2Bcu100.zip
libtorch-shared-with-deps-1.4.0.zip : https://download.pytorch.org/libtorch/cu101/libtorch-shared-with-deps-1.4.0.zip
libtorch-shared-with-deps-1.5.0+cu92.zip : https://download.pytorch.org/libtorch/cu92/libtorch-shared-with-deps-1.5.0%2Bcu92.zip
libtorch-shared-with-deps-1.5.0+cu101.zip : https://download.pytorch.org/libtorch/cu101/libtorch-shared-with-deps-1.5.0%2Bcu101.zip
libtorch-shared-with-deps-1.5.0.zip : https://download.pytorch.org/libtorch/cu101/libtorch-shared-with-deps-1.5.0.zip
libtorch-shared-with-deps-1.5.0+cu102.zip : https://download.pytorch.org/libtorch/cu102/libtorch-shared-with-deps-1.5.0%2Bcu102.zip
libtorch-shared-with-deps-1.5.0.zip : https://download.pytorch.org/libtorch/cu102/libtorch-shared-with-deps-1.5.0.zip
libtorch-shared-with-deps-1.5.1+cu92.zip : https://download.pytorch.org/libtorch/cu92/libtorch-shared-with-deps-1.5.1%2Bcu92.zip
libtorch-shared-with-deps-1.5.1+cu101.zip : https://download.pytorch.org/libtorch/cu101/libtorch-shared-with-deps-1.5.1%2Bcu101.zip
libtorch-shared-with-deps-1.5.1.zip : https://download.pytorch.org/libtorch/cu102/libtorch-shared-with-deps-1.5.1.zip
libtorch-shared-with-deps-1.6.0+cu92.zip : https://download.pytorch.org/libtorch/cu92/libtorch-shared-with-deps-1.6.0%2Bcu92.zip
libtorch-shared-with-deps-1.6.0+cu101.zip : https://download.pytorch.org/libtorch/cu101/libtorch-shared-with-deps-1.6.0%2Bcu101.zip
libtorch-shared-with-deps-1.6.0.zip : https://download.pytorch.org/libtorch/cu102/libtorch-shared-with-deps-1.6.0.zip
libtorch-shared-with-deps-1.7.0+cu92.zip : https://download.pytorch.org/libtorch/cu92/libtorch-shared-with-deps-1.7.0%2Bcu92.zip
libtorch-shared-with-deps-1.7.0+cu101.zip : https://download.pytorch.org/libtorch/cu101/libtorch-shared-with-deps-1.7.0%2Bcu101.zip
libtorch-shared-with-deps-1.7.0.zip : https://download.pytorch.org/libtorch/cu102/libtorch-shared-with-deps-1.7.0.zip
libtorch-shared-with-deps-1.7.0+cu110.zip : https://download.pytorch.org/libtorch/cu110/libtorch-shared-with-deps-1.7.0%2Bcu110.zip
libtorch-shared-with-deps-1.7.1+cu92.zip : https://download.pytorch.org/libtorch/cu92/libtorch-shared-with-deps-1.7.1%2Bcu92.zip
libtorch-shared-with-deps-1.7.1+cu101.zip : https://download.pytorch.org/libtorch/cu101/libtorch-shared-with-deps-1.7.1%2Bcu101.zip
libtorch-shared-with-deps-1.7.1.zip : https://download.pytorch.org/libtorch/cu102/libtorch-shared-with-deps-1.7.1.zip
libtorch-shared-with-deps-1.7.1+cu110.zip : https://download.pytorch.org/libtorch/cu110/libtorch-shared-with-deps-1.7.1%2Bcu110.zip
libtorch-shared-with-deps-1.8.0+cu101.zip : https://download.pytorch.org/libtorch/cu101/libtorch-shared-with-deps-1.8.0%2Bcu101.zip
libtorch-shared-with-deps-1.8.0.zip : https://download.pytorch.org/libtorch/cu102/libtorch-shared-with-deps-1.8.0.zip
libtorch-shared-with-deps-1.8.0+cu111.zip : https://download.pytorch.org/libtorch/cu111/libtorch-shared-with-deps-1.8.0%2Bcu111.zip
libtorch-shared-with-deps-1.8.0+cu112.zip : https://download.pytorch.org/libtorch/cu112/libtorch-shared-with-deps-1.8.0%2Bcu112.zip
libtorch-shared-with-deps-1.8.1+cu101.zip : https://download.pytorch.org/libtorch/cu101/libtorch-shared-with-deps-1.8.1%2Bcu101.zip
libtorch-shared-with-deps-1.8.1+cu102.zip : https://download.pytorch.org/libtorch/cu102/libtorch-shared-with-deps-1.8.1%2Bcu102.zip
libtorch-shared-with-deps-1.8.1+cu111.zip : https://download.pytorch.org/libtorch/cu111/libtorch-shared-with-deps-1.8.1%2Bcu111.zip
libtorch-shared-with-deps-1.9.0+cu102.zip : https://download.pytorch.org/libtorch/cu102/libtorch-shared-with-deps-1.9.0%2Bcu102.zip
libtorch-shared-with-deps-1.9.0+cu111.zip : https://download.pytorch.org/libtorch/cu111/libtorch-shared-with-deps-1.9.0%2Bcu111.zip
libtorch-shared-with-deps-1.10.0+cu102.zip : https://download.pytorch.org/libtorch/cu102/libtorch-shared-with-deps-1.10.0%2Bcu102.zip
libtorch-shared-with-deps-1.10.0+cu111.zip : https://download.pytorch.org/libtorch/cu111/libtorch-shared-with-deps-1.10.0%2Bcu111.zip
libtorch-shared-with-deps-1.10.0+cu113.zip : https://download.pytorch.org/libtorch/cu113/libtorch-shared-with-deps-1.10.0%2Bcu113.zip
libtorch-shared-with-deps-1.10.1+cu102.zip : https://download.pytorch.org/libtorch/cu102/libtorch-shared-with-deps-1.10.1%2Bcu102.zip
libtorch-shared-with-deps-1.10.1+cu111.zip : https://download.pytorch.org/libtorch/cu111/libtorch-shared-with-deps-1.10.1%2Bcu111.zip
libtorch-shared-with-deps-1.10.1+cu113.zip : https://download.pytorch.org/libtorch/cu113/libtorch-shared-with-deps-1.10.1%2Bcu113.zip

```

# Win libtorch 1.4.0 ~ 1.10.1 gpu version url

**Win libtorch url is similar to Linux's *Pre-cxx11 ABI* , such as libtorch 1.10.1:**

- Linux libtorch 1.10.1(Pre-cxx11 ABI)

CUDA 10.2 : https://download.pytorch.org/libtorch/cu102/libtorch-shared-with-deps-1.10.1%2Bcu102.zip 
CUDA 11.3 : https://download.pytorch.org/libtorch/cu113/libtorch-shared-with-deps-1.10.1%2Bcu113.zip

- Linux libtorch 1.10.1(Release version)

CUDA 10.2 : https://download.pytorch.org/libtorch/cu102/libtorch-win-shared-with-deps-1.10.1%2Bcu102.zip
CUDA 11.3 : https://download.pytorch.org/libtorch/cu113/libtorch-win-shared-with-deps-1.10.1%2Bcu113.zip

- Linux libtorch 1.10.1(Debug version)

CUDA 10.2 : https://download.pytorch.org/libtorch/cu102/libtorch-win-shared-with-deps-debug-1.10.1%2Bcu102.zip
CUDA 11.3 : https://download.pytorch.org/libtorch/cu113/libtorch-win-shared-with-deps-debug-1.10.1%2Bcu113.zip

**All others Win libtorch url is similar above !**

# Linux and Win libtorch cpu version url

Replace **cu113** (or **cu102**, **cu101**, ......) with **cpu** in gpu url and then you can get cpu url, such as:

gpu : https://download.pytorch.org/libtorch/cu113/libtorch-shared-with-deps-1.10.1%2Bcu113.zip # (CUDA 11.3) 
cpu : https://download.pytorch.org/libtorch/cpu/libtorch-shared-with-deps-1.10.1%2Bcpu.zip # (CUDA 11.3)

# All pytorch version is here

all pytorch version : https://download.pytorch.org/whl/torch_stable.html

**Comment by patrikhuber** [comment]
@peterjc123 Well... I needed libtorch 1.13.1 binaries for Windows. Went here: https://[download.pytorch.org/libtorch/cu117](https://download.pytorch.org/libtorch/cu117) - shows a nice directory listing with all the files I need (`[libtorch-win-shared-with-deps-1.13.1%2Bcu117.zip](https://download.pytorch.org/libtorch/libtorch-win-shared-with-deps-1.13.1%2Bcu117.zip)`, `[libtorch-win-shared-with-deps-debug-1.13.1%2Bcu117.zip](https://download.pytorch.org/libtorch/libtorch-win-shared-with-deps-debug-1.13.1%2Bcu117.zip)`). However when you click them - "Access Denied" error, the files don't seem available anymore:
![image](https://github.com/pytorch/pytorch/assets/4967343/c63c2d64-270c-496b-aa51-fa34927a5529)

Presumably the files might still be there, just the access denied... anyway to restore these? It would be invaluable.

**Comment by patrikhuber** [comment]
> Are you looking for libtorch-win-shared-with-deps-debug-1.13.1?

Well, yes and no. I was really looking for CUDA builds, not CPU builds. It's a shame those are not available anymore.

I solved it in the end by upgrading to libtorch-2.1.1, luckily my code didn't need any changes.

**Comment by linClubs** [comment]
> > Are you looking for libtorch-win-shared-with-deps-debug-1.13.1?
> 
> Well, yes and no. I was really looking for CUDA builds, not CPU builds. It's a shame those are not available anymore.
> 
> I solved it in the end by upgrading to libtorch-2.1.1, luckily my code didn't need any changes.

---

Up to present, U can download libtorch from the follow url. 

[libtorch-cuda12.1](https://download.pytorch.org/libtorch/cu121)
[libtorch-cuda11.8](https://download.pytorch.org/libtorch/cu118)
[libtorch-cuda11.3](https://download.pytorch.org/libtorch/cu113)
[libtorch-cuda10.1](https://download.pytorch.org/libtorch/cu101)
[libtorch-cpu](https://download.pytorch.org/libtorch/cpu)

**Comment by patrikhuber** [comment]
> Up to present, U can download libtorch from the follow url.
> 
> [libtorch-cuda12.1](https://download.pytorch.org/libtorch/cu121) [libtorch-cuda11.8](https://download.pytorch.org/libtorch/cu118) [libtorch-cuda11.3](https://download.pytorch.org/libtorch/cu113) [libtorch-cuda10.1](https://download.pytorch.org/libtorch/cu101) [libtorch-cpu](https://download.pytorch.org/libtorch/cpu)

No, you can't ;-). This list of releases looks great, until you click one of them, and get an error:

![image](https://github.com/pytorch/pytorch/assets/4967343/7fdc21fa-6ca4-40ed-a70a-eb27894bcd22)

As I said in my earlier post, unfortunately these releases are not available anymore.

Anyway, I solved my problem by upgrading to 2.x.

**Comment by sullivak** [comment]
Is there a solution if not upgrading?

**Comment by samueldyoung29ctr** [comment]
For anyone looking for URLs more recently, as of 08 Jul 2024 the URL `https://download.pytorch.org/libtorch/cu<your_CUDA_version>/` appears to show a list of all libtorch versions available for that CUDA version. E.g., https://download.pytorch.org/libtorch/cu118 shows all variants available for CUDA 11.8.
